### ISSUE TITLE
Cypress component tests not working out of the box

### ISSUE THREAD

**Issue by gilian-rockstars** [nrwl/nx#12023]
Component tests doesn't seem to work out of the box.

## Current Behavior
1. Create new NX Angular project. `npx create-nx-workspace@latest`
2. Generate component-tests. `nx g @nrwl/angular:cypress-component-configuration --project=your-project --generate-tests`
3. Run tests. `nx component-test your-project`
4. Get error:
```
ERROR in ./cypress/support/component.ts
Module build failed (from ../../node_modules/@ngtools/webpack/src/ivy/index.js):
Error: /Users/user/Documents/Temp/component-test/apps/yolo/cypress/support/component.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at /Users/user/Documents/Temp/component-test/node_modules/@ngtools/webpack/src/ivy/loader.js:64:26

```

## Expected Behavior
1. Create new NX Angular project. `npx create-nx-workspace@latest`.
2. Generate component-tests. `nx g @nrwl/angular:cypress-component-configuration --project=your-project --generate-tests`.
3. Run tests. `nx component-test your-project`.
4. Component tests work.

## Steps to Reproduce
See 'Current behaviour'.

### Failure Logs
```
ERROR in ./cypress/support/component.ts
Module build failed (from ../../node_modules/@ngtools/webpack/src/ivy/index.js):
Error: /Users/user/Documents/Temp/component-test/apps/yolo/cypress/support/component.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at /Users/user/Documents/Temp/component-test/node_modules/@ngtools/webpack/src/ivy/loader.js:64:26

ERROR in ./src/app/app.component.cy.ts
Module build failed (from ../../node_modules/@ngtools/webpack/src/ivy/index.js):
Error: /Users/user/Documents/Temp/component-test/apps/yolo/src/app/app.component.cy.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at /Users/user/Documents/Temp/component-test/node_modules/@ngtools/webpack/src/ivy/loader.js:64:26

ERROR in ./src/app/nx-welcome.component.cy.ts
Module build failed (from ../../node_modules/@ngtools/webpack/src/ivy/index.js):
Error: /Users/user/Documents/Temp/component-test/apps/yolo/src/app/nx-welcome.component.cy.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at /Users/user/Documents/Temp/component-test/node_modules/@ngtools/webpack/src/ivy/loader.js:64:26

```

### Environment
Just the defaults when creating a new project.

**Comment by gilian-rockstars** [comment]
Quick addition: if component tests are started with the `--watch` tag, you can "run" a test, however cypress will be stuck with the message "Your tests are loading...".

**Comment by yusijs** [comment]
I outlined a bit about this in this comment:
https://github.com/nrwl/nx/issues/11602#issuecomment-1247836881

tl;dr tsconfig.app.json ignores *.cy.(ts|tsx|js) and cypress/, nrwl/cypress tells cypress to use tsconfig.app.json. However, cypress ignores the tsconfig passed in, but there is a fix for that in cypress that should come out with the next release (seems like a bi-weekly release schedule). 

I'm also having issues with scss-compilation (doesn't work with external scss files from node_modules, even if stylePreprocessorOptions.includePaths = ["node_modules"].

So I agree, component-tests are still lacking..

**Comment by barbados-clemens** [comment]
fixed in #12014
